modules/globebrowsing/util/globelabelscomponent.cpp

```diff
@@ -0,0 +1,842 @@
+            "GlobeLabelsComponent|" + identifier(),
+            ghoul::filesystem::CacheManager::Persistent::Yes
+        );
+
+        bool hasCachedFile = FileSys.fileExists(cachedFile);
+        if (hasCachedFile) {
+            LINFO(fmt::format(
+                "Cached file '{}' used for labels file '{}'",
+                cachedFile,
+                file
+            ));
+
+            success = loadCachedFile(cachedFile);
+            if (success) {
+                return true;
+            }
+            else {
+                FileSys.cacheManager()->removeCacheFile(file);
+                // Intentional fall-through to the 'else' to generate the cache
+                // file for the next run
+            }
+        }
+        else {
+            LINFO(fmt::format("Cache for labels file '{}' not found", file));
+        }
+        LINFO(fmt::format("Loading labels file '{}'", file));
+
+        success = readLabelsFile(file);
+        if (!success) {
+            return false;
+        }
+
+        success &= saveCachedFile(cachedFile);
+    }
+    return success;
+}
+
+bool GlobeLabelsComponent::readLabelsFile(const std::string& file) {
+    try {
+        std::fstream csvLabelFile(file);
+        if (!csvLabelFile.good()) {
+            LERROR(fmt::format("Failed to open labels file '{}'", file));
+            return false;
+        }
+        if (csvLabelFile.is_open()) {
+            char line[4096];
+            _labels.labelsArray.clear();
+            while (!csvLabelFile.eof()) {
+                csvLabelFile.getline(line, 4090);
+                if (strnlen(line, 4090) > 10) {
+                    LabelEntry lEntry;
+                    char *token = strtok(line, ",");
+                    // First line is just the Header
+                    if (strcmp(token, "Feature_Name") == 0) {
+                        continue;
+                    }
+                    strncpy(lEntry.feature, token, 256);
+                    // Removing non ASCII characters:
+                    int tokenChar = 0;
+                    while (tokenChar < 256) {
+                        if ((lEntry.feature[tokenChar] < 0 ||
+                            lEntry.feature[tokenChar] > 127) &&
+                            lEntry.feature[tokenChar] != '\0') {
+                            lEntry.feature[tokenChar] = '*';
+                        }
+                        else if (lEntry.feature[tokenChar] == '\"') {
+                            lEntry.feature[tokenChar] = '=';
+                        }
+                        tokenChar++;
+                    }
+
+                    strtok(NULL, ","); // Target is not used
+                    lEntry.diameter = static_cast<float>(atof(strtok(NULL, ",")));
+                    lEntry.latitude = static_cast<float>(atof(strtok(NULL, ",")));
+                    lEntry.longitude = static_cast<float>(atof(strtok(NULL, ",")));
+                    char * coordinateSystem = strtok(NULL, ",");
```

Comment:
A lot of c-style string functions being used here. Looks like c++ streams could be used to make the code more easy to follow, and match the rest of the coding style. Any particular reason to use the c-style functions?

Reply:
I have to disagree with you on this point. Strtok can be used much easier here yielding a more readable code than the equivalent code with streams. It is true that it may not match the rest of the coding style but I think that is a small price to pay for readability.